### PR TITLE
210 restructure photolysis config objects

### LINKED ISSUE #210: Restructure photolysis config objects
Make the configuration for photolysis rates be an array instead of an object of objects

Update the documentation if necessary

### PR DESCRIPTION
**Pull request by K20shores**
closes #210 

[updated docs](https://ncar.github.io/tuv-x/branch/210-restructure-photolysis-config-objects/configuration/#photolysis-reactions)

### CODE REVIEW

File: src/photolysis_rates.F90 [hunk 1]
```diff
@@ -102,10 +102,7 @@ function constructor( reaction_set, grid_warehouse, profile_warehouse )     &
     allocate( rates%quantum_yields_(0) )
     allocate( rates%scaling_factors_(0) )
 
-    ! iterate over photo reactions
-    iter => reaction_set%get_iterator( )
-
-    call reaction_set%get( "enable diagnostics",                              &
+    call photolysis_config%get( "enable diagnostics",                         &
```
Comment (mattldawson) [inline src/photolysis_rates.F90:105]: we could use the config data validation now that there are no user defined keys. Similar to here: https://github.com/NCAR/tuv-x/blob/f1c699c1e2a0f6355018767471aa4d3ed5382703/src/core.F90#L89-L99

**Comment by mattldawson** [comment]
did you mean to update `musica-core`?

**Comment by K20shores** [comment]
> did you mean to update `musica-core`?

No, I really don't understand why that happened.

File: src/photolysis_rates.F90 (hunk 1 above)
Comment (K20shores) [reply src/photolysis_rates.F90:105]: oh good point, I'll add that

**Review by mattldawson — APPROVED** [review]

**Pull request merged**